File: Python/Number of Boomerangs.py
class Solution(object):
    def numberOfBoomerangs(self, points):
        """
        :type points: List[List[int]]
        :rtype: int
        """
        result = 0
        for point in points:
            dis_dic = {}
            for point_b in points:
                dis = self.compute_dis(point, point_b)
                if dis in dis_dic:
                    dis_dic[dis] += 1
                else:
                    dis_dic[dis] = 1
            for key in dis_dic:
                result += dis_dic[key] * (dis_dic[key] - 1)
        return result
                
    
    
    def compute_dis(self, a, b):
        return (a[0] - b[0]) ** 2 + (a[1] - b[1]) ** 2
                
            

File: cpp/Number of Boomerangs.cpp
class Solution {
public:
    int numberOfBoomerangs(vector<pair<int, int>>& points) {
        int result = 0;
        for (auto point : points) {
            map<int, int> dis_dic;
            for (auto point_b : points) {
                int dis = compute_dis(point, point_b);
                if (dis_dic.find(dis) != dis_dic.end()) {
                    dis_dic[dis]++;
                } else {
                    dis_dic[dis] = 1;
                }
            }
            for (auto p : dis_dic) {
                result += p.second * (p.second - 1);
            }
        }
        return result;
    }

    int compute_dis(pair<int, int> a, pair<int, int> b) {
        return (a.first - b.first) * (a.first - b.first) + (a.second - b.second) * (a.second - b.second);
    }
};
